controlplane/src/workers/compose.ts

```diff
@@ -0,0 +1,60 @@
+/**
+  * This file is intended to be used with a worker thread.
+  * In watch mode changes aren't immediately applied. You have to rebuild the project with tsc before changes are applied.
+  */
+import {
+  ContractTagOptions,
+  FederationResult,
+  FederationResultWithContracts,
+  newContractTagOptionsFromArrays,
+} from '@wundergraph/composition';
+import { composeFederatedContract, composeFederatedGraphWithPotentialContracts } from '../core/composition/composition.js';
+import {
+  CompositionOptions,
+  FederatedGraphDTO,
+} from '../types/index.js';
+import { SubgraphsToCompose } from '../core/repositories/FeatureFlagRepository.js';
+import { Tinypool } from 'tinypool';
+import { isMainThread } from 'worker_threads';
+import { join } from 'path';
+
+interface Inputs {
+  federatedGraph: FederatedGraphDTO;
+  subgraphsToCompose: SubgraphsToCompose;
+  tagOptionsByContractName: Map<string, ContractTagOptions>;
+  compositionOptions?: CompositionOptions;
+}
+
+export function getWorkerPool(maxCount?: number): Tinypool | undefined {
+  if (isMainThread) {
+    const filename = import.meta.url.endsWith('.ts') ?
```

Comment:
What's the purpose of this? In all cases, we need to point to `dist/workers/compose.js` right?